### DISCUSSION TITLE
No turbulent inflow fluctuations occur

### DISCUSSION THREAD

**Discussion by sjlienge** [AMReX-Combustion/PeleC#717] [Q&A] [answered]
Dear PeleC community,

In the Cavity Flame I wanted to test the inflow turbulence capability of PeleC. But I cannot observe any inflow turbulence with the settings I use.

From the PeleLMeX Turbinflow I have generated the TurbFileHIT using ./python3 -k0 2 -N 64 and u_rms = 1.0 and copied it into my PeleC run-directory.

I have adjusted the settings following this discussion:
https://github.com/AMReX-Combustion/PeleC/discussions/656

```
### BOUNDARY FLAGS
pelec.lo_bc = "Hard"	 "Hard"		"Interior"
pelec.hi_bc = "FOExtrap" "NoSlipWall"	"Interior"

### TURBULENT INFLOW
turbinflows = lowX
turbinflow.lowX.turb_file      = TurbFileHIT/TurbTEST
# the inflow has to be in x direction
turbinflow.lowX.dir            = 0
# inflow at lower X boundary
turbinflow.lowX.side           = low
# 2*pi/(1.5 cm) 1.5 is the height of the inflow channel
turbinflow.lowX.turb_scale_loc = 4.19
# I used a high value for now in the hope to see any kind of fluctuations
turbinflow.lowX.turb_scale_vel = 1500.0
# Mean inflow velocity (same as the bulk velocity of the inflow I prescribed)
turbinflow.lowX.turb_conv_vel  = 3169.4
# I wasnt sure here: y and z coordinates of the center of the inflow area?
turbinflow.lowX.turb_center    = 1.05 0.0
# cells in each direction as in TurbTEST
turbinflow.lowX.turb_nplane    = 64
```

In the prob.H in the bcnormal I have adjusted
`s_ext[UMX] = rho * vx_in `
to 
`s_ext[UMX] += rho * vx_in`

The calculation runs but even after a significant amount of run time there is no fluctuations in the x, y or z velocities observable. What am I doing wrong?
Thank you in advance already :)

**Comment by marchdf** [comment, marked as answer]
This is most likely a units problem (MKS vs CGS). PeleC is CGS and LMeX is MKS. So we need to scale the data. We should document this better so I opened an issue for us in pelephysics to fix it (https://github.com/AMReX-Combustion/PelePhysics/issues/455). In the mean time, you can see how we do it in PeleC here: https://github.com/AMReX-Combustion/PeleC/blob/development/Exec/Production/ChallengeProblem/challenge.inp#L115. I hope this helps!

**Reply by sjlienge** [reply]
Thank you so much for the quick response!

I looked at the ChallengeProblem and it was a small thing that needed to be changed. In the prob.H I changed
const amrex::GpuArray<amrex::Real, AMREX_SPACEDIM>& /*turb_fluc*/)
to
const amrex::GpuArray<amrex::Real, AMREX_SPACEDIM>& turb_fluc)
and now it seems to work :)

**Reply by marchdf** [reply]
glad it was so easy!

**Reply by Pockjob** [reply]
Hi @sjlienge @marchdf , It's interesting to set turbulent inlet in the cavity flame. I want to test the same case but I have no tool to get the TurbFileHIT/TurbTEST file, and I am very curious about how to modify the bcnormal ().  If it's possibile, could you upload your case for me to learn it? I'd appreciate your help.

**Reply by baperry2** [reply]
The utility to generate these files exists within PeleLMeX: https://github.com/AMReX-Combustion/PeleLMeX/tree/development/Exec/RegTests/TurbInflow/TurbFileHIT

There's a short README in that directory with instructions on how to generate these files.

For an example of how to modify `bcnormal` in PeleC, see the implementation here: https://github.com/AMReX-Combustion/PeleC/blob/237b9c8ff6c6f6dbfb049db0c22bfc3c6a9ee3fd/Exec/Production/ChallengeProblem/prob.H#L248

Basically, the `turb_fluc` GpuArray contains the three components of turbulent velocity fluctuations at the local point in space/time, and you can do whatever you want with that within bcnormal.

**Reply by Pockjob** [reply]
Thanks @baperry2, I will test it.
